contrib/IECoreAlembic/src/IECoreAlembic/AlembicScene.cpp

```diff
@@ -404,16 +428,40 @@ class AlembicScene::AlembicReader : public AlembicIO
 			}
 
 			ICompoundProperty userProperties = m_xform.getSchema().getUserProperties();
-			const PropertyHeader *propertyHeader = userProperties.getPropertyHeader( name.string() );
 
-			if( !propertyHeader )
+			if ( !userProperties.valid() )
+			{
+				return 0;
+			}
+
+			Abc::CompoundPropertyReaderPtr propertyReader = GetCompoundPropertyReaderPtr( userProperties );
+
+			if ( !propertyReader )
 			{
 				return 0.0;
 			}
 
-			auto timeSampling = propertyHeader->getTimeSampling();
+			ScalarPropertyReaderPtr scalarPropertyReader = propertyReader->getScalarProperty( name.string() );
+
+			if ( !scalarPropertyReader )
+			{
+				return 0.0;
+			}
+
+			std::pair<index_t, chrono_t> floor = scalarPropertyReader->getFloorIndex( time );
```

Comment:
Can we continue to use our `sampleInterval()` utility for this as we do for all the other interval queries? I'd be happier if we can, since we have some heuristics to avoid doing unnecessary interpolation when we're so near-as-dammit on a sample that it would just represent pointless overhead. It's also less code to maintain. From a quick squint at the headers it looks like we have the time sampling and number of samples available to us via scalarPropertyReader...

Reply:
Good point updated in b931144. @johnhaddon happy for me to squash & merge?

Reply:
Yupyup - thanks Don!